foundationdb/src/tuple/element.rs

```diff
@@ -312,6 +323,17 @@ impl Decode for Vec<Element> {
             buf = &buf[offset..];
         }
 
+        if nested && tuple_depth.depth() == 0 && buf.len() > 0 {
```

Comment:
Nested already checks for but.len() > 0, no?

Reply:
same as for your previous comment, buf is shrunk by the loop